tests/node_api.rs

```diff
@@ -2,293 +2,265 @@
+    Client::builder()
+        .with_node(DEFAULT_DEVNET_NODE_URL)
         .unwrap()
         .with_node_sync_disabled()
         .finish()
         .await
-        .unwrap();
+        .unwrap()
+}
+
+// Sends a tagged data message to the node to test against it.
+async fn setup_tagged_data_message() -> MessageId {
+    let client = setup_client_with_sync_disabled().await;
 
     client
         .message()
-        .with_tag("iota.rs")
-        .with_data("iota.rs".as_bytes().to_vec())
+        .with_tag("Hello")
+        .with_data("Tangle".as_bytes().to_vec())
         .finish()
         .await
         .unwrap()
         .id()
 }
 
-const DEFAULT_NODE_POOL_URLS: &str = "https://giftiota.com/nodes.json";
-#[tokio::test]
-#[ignore]
-async fn test_with_node_pool_urls() {
-    let r = iota_client::Client::builder()
-        .with_node_pool_urls(&[DEFAULT_NODE_POOL_URLS.into()])
+async fn setup_secret_manager() -> SecretManager {
+    SecretManager::Mnemonic(MnemonicSecretManager::try_from_hex_seed(DEFAULT_DEVELOPMENT_SEED).unwrap())
+}
+
+// Sends a transaction message to the node to test against it.
+async fn setup_transaction_message() -> (MessageId, TransactionId) {
+    let client = setup_client_with_sync_disabled().await;
+    let secret_manager = setup_secret_manager().await;
+
+    let addresses = client
+        .get_addresses(&secret_manager)
+        .with_range(0..2)
+        .get_raw()
+        .await
+        .unwrap();
+    println!(
+        "{}",
+        request_funds_from_faucet(
+            DEFAULT_DEVNET_FAUCET_URL,
+            &addresses[0].to_bech32(client.get_bech32_hrp().await.unwrap()),
+        )
         .await
         .unwrap()
+    );
+    tokio::time::sleep(std::time::Duration::from_secs(20)).await;
+
+    let message_id = client
+        .message()
+        .with_secret_manager(&secret_manager)
+        .with_output_hex(
+            &bech32_to_hex(&addresses[1].to_bech32(client.get_bech32_hrp().await.unwrap())).unwrap(), /* Send funds
+                                                                                                       * back to the
+                                                                                                       * sender. */
+            1_000_000, // The amount to spend, cannot be zero.
+        )
+        .unwrap()
         .finish()
-        .await;
-    println!("{:#?}", r);
-}
+        .await
+        .unwrap()
+        .id();
 
-#[tokio::test]
-#[ignore]
-async fn test_get_info() {
-    let r = iota_client::Client::get_node_info(DEFAULT_NODE_URL, None)
+    let message = setup_client_with_sync_disabled()
+        .await
+        .get_message_data(&message_id)
         .await
         .unwrap();
-    println!("{:#?}", r);
+
+    let transaction_id = match message.payload() {
+        Some(Payload::Transaction(t)) => t.id(),
+        _ => unreachable!(),
+    };
+
+    tokio::time::sleep(std::time::Duration::from_secs(20)).await;
```

Comment:
Here I would rather use `let _ = client.retry_until_included(&message.id(), None, None).await?;`, because we know the message, only for the faucet request we don't know it

Reply:
Fixed it